AgentgatewayParameters overlays explained (#123)
agentgateway/website: assets/agw-docs/pages/agentgateway/configuration.md

kind: {{< reuse "agw-docs/snippets/gatewayparameters.md" >}}
metadata:
  name: agentgateway-openshift
  namespace: {{< reuse "agw-docs/snippets/namespace.md" >}}
spec:
  deployment:
    spec:
      template:
        spec:
          # Delete pod-level securityContext using $patch: delete (works with any apply mode)
          securityContext:
            $patch: delete
          containers:
            - name: agentgateway
              # Delete container-level securityContext using null (requires server-side apply)
              securityContext: null
EOF
```

### Add environment variables {#recipe-env-vars}

Add custom environment variables to the agentgateway container. Use `null` to remove default environment variables.

```yaml
kubectl apply --server-side -f- <<'EOF'
apiVersion: {{< reuse "agw-docs/snippets/gatewayparam-apiversion.md" >}}
kind: {{< reuse "agw-docs/snippets/gatewayparameters.md" >}}
metadata:
  name: agentgateway-env
  namespace: {{< reuse "agw-docs/snippets/namespace.md" >}}
spec:
  env:
    - name: MY_CUSTOM_VAR
      value: "my-value"
    - name: CONNECTION_MIN_TERMINATION_DEADLINE
      value: "500s"

>     # Remove a default env var by setting value to null

Reviewer: ```suggestion
    # Remove a default env var by setting the value to null
```

kind: {{< reuse "agw-docs/snippets/gatewayparameters.md" >}}
metadata:
  name: agentgateway-hpa
  namespace: {{< reuse "agw-docs/snippets/namespace.md" >}}
spec:
  horizontalPodAutoscaler:
    metadata:
      labels:
        app: agentgateway
    spec:
      minReplicas: 2
      maxReplicas: 10
      metrics:
        - type: Resource
          resource:
            name: cpu
            target:
              type: Utilization
              averageUtilization: 80
EOF
```

### Set up Pod Disruption Budget (PDB) {#recipe-pdb}

Configure a Pod Disruption Budget to ensure availability during voluntary disruptions. The PDB resource is only created when you specify this overlay.

```yaml
kubectl apply --server-side -f- <<'EOF'
apiVersion: {{< reuse "agw-docs/snippets/gatewayparam-apiversion.md" >}}
kind: {{< reuse "agw-docs/snippets/gatewayparameters.md" >}}
metadata:
  name: agentgateway-pdb
  namespace: {{< reuse "agw-docs/snippets/namespace.md" >}}
spec:
  podDisruptionBudget:
    metadata:
      labels:
        app: agentgateway
    spec:
      minAvailable: 1
EOF
```

### Mount custom ConfigMap as volume {#recipe-configmap-volume}

> Mount a custom ConfigMap into the agentgateway container. This example replaces the default volumes and adds a custom config.

Reviewer: ```suggestion
Mount a custom ConfigMap to the `agentgateway` container that runs inside your agentgateway proxy pod. This example replaces the default volumes with a custom config.
```

apiVersion: {{< reuse "agw-docs/snippets/gatewayparam-apiversion.md" >}}
kind: {{< reuse "agw-docs/snippets/gatewayparameters.md" >}}
metadata:
  name: agentgateway-labels
  namespace: {{< reuse "agw-docs/snippets/namespace.md" >}}
spec:
  deployment:
    metadata:
      labels:
        environment: production
        team: platform
      annotations:
        description: "Production agentgateway proxy"
    spec:
      template:
        metadata:
          labels:
            environment: production
          annotations:
            prometheus.io/scrape: "true"
            prometheus.io/port: "15020"
  service:
    metadata:
      annotations:
        service.beta.kubernetes.io/aws-load-balancer-type: "nlb"
EOF
```

### Customize service ports {#recipe-service-ports}

Replace the default service ports with custom port configurations.

```yaml
kubectl apply --server-side -f- <<'EOF'
apiVersion: {{< reuse "agw-docs/snippets/gatewayparam-apiversion.md" >}}
kind: {{< reuse "agw-docs/snippets/gatewayparameters.md" >}}
metadata:
  name: agentgateway-service-ports
  namespace: {{< reuse "agw-docs/snippets/namespace.md" >}}
spec:
  service:
    spec:
      ports:

>         - $patch: replace

Reviewer: what does this line do? does it not just overwrite it automatically?

        - name: http
          port: 80
          targetPort: 8080
          protocol: TCP
        - name: https
          port: 443
          targetPort: 8443
          protocol: TCP
EOF
```

### Set resource requests and limits {#recipe-resources}

Configure CPU and memory requests and limits for the agentgateway container.

```yaml
kubectl apply --server-side -f- <<'EOF'
apiVersion: {{< reuse "agw-docs/snippets/gatewayparam-apiversion.md" >}}
kind: {{< reuse "agw-docs/snippets/gatewayparameters.md" >}}
metadata:
  name: agentgateway-resources
  namespace: {{< reuse "agw-docs/snippets/namespace.md" >}}
spec:
  resources:
    requests:
      cpu: 100m
      memory: 128Mi
    limits:
      cpu: 500m
      memory: 512Mi

> EOF

Reviewer: how does it know it will get applied to the agentgateway container without a matchlabel? is that because it is the only container currently in the proxy?

```

### Custom pod security context {#recipe-security-context}

Configure custom security settings for the pod and containers.

```yaml
kubectl apply --server-side -f- <<'EOF'
apiVersion: {{< reuse "agw-docs/snippets/gatewayparam-apiversion.md" >}}
kind: {{< reuse "agw-docs/snippets/gatewayparameters.md" >}}
metadata:
  name: agentgateway-security
  namespace: {{< reuse "agw-docs/snippets/namespace.md" >}}
spec:
  deployment:
    spec:
      template:
        spec:
          securityContext:
            runAsUser: 1000
            runAsGroup: 2000
            fsGroup: 3000
EOF
```

> ### Remove a label with null {#recipe-remove-label}

Reviewer: ```suggestion
### Remove default labels {#recipe-remove-label}
```
